feat(createRetryFetch): adding an enhanced retry fetch (#18)
americanexpress/fetch-enhancers: README.md

const fastRequest = timeoutFetch('https://example.com/fast', { timeout: 1e3 });
```

### createRetryFetch [Server & Browser]
```js
import { createRetryFetch } from '@americanexpress/fetch-enhancers';
```

#### Configuring

> `createRetryFetch` has three arguments: another enhanced fetch (required), an *optional* retry count, and an *optional* back off strategy function:

Reviewer: ```suggestion
`createRetryFetch` has two arguments: an *optional* retry count and an *optional* back off strategy function:
```

```js

> const enhancedRetryFetch = createRetryFetch(enhancedTimeoutFetch)(fetch);

Reviewer: ```suggestion
const enhancedRetryFetch = createRetryFetch()(fetch);
```

```

Optional retry count and back off strategy function that accepts the current retry count:
```js

> const enhancedRetryFetch = createRetryFetch(enhancedTimeoutFetch, 3,
>   (n) => new Promise((res) => setTimeout(res, n * 1000)))(fetch);

Reviewer: maybe:
```suggestion
const retryAfterTimeoutsFetch = createRetryFetch(3, (n) => new Promise((res) => setTimeout(res, n * 1000)))(
  enhancedTimeoutFetch()(
    fetch
  )
);
```

const fastRequest = timeoutFetch('https://example.com/fast', { timeout: 1e3 });
```

### createRetryFetch [Server & Browser]
```js
import { createRetryFetch } from '@americanexpress/fetch-enhancers';
```

#### Configuring
`createRetryFetch` accepts the an object with the following configuration: an *optional* maxRetry (default 3), and an *optional* back off strategy function:

```js
const enhancedRetryFetch = createRetryFetch()(fetch);
```

Optional retry count and back off strategy function that accepts the current retry count:
```js
const enhancedRetryFetch = createRetryFetch({
  maxRetry: 5,
  backoffStrategy: (n) => new Promise((res) => setTimeout(res, n * 1000)),
})(fetch);
```

#### Example

```js
import { createRetryFetch } from '@americanexpress/fetch-enhancers';

const retryFetch = createRetryFetch()(createTimeoutFetch(5e3)(fetch));

// Then use retryFetch as you would normally

> const request = retryFetch('https://example.com');
> request.then((response) => response.json())

Reviewer: might be more common to
```suggestion
retryFetch('https://example.com')
  .then((response) => response.json())
```
but not a blocker

Author: Agree. I was trying to keep it consistent with line #92
